### PR TITLE
r.niche.similarity: address #897

### PR DESCRIPTION
**Pull request by veroandreo**
This small PR solves the errors reported in #897. The printing in the csv file does not look great though... not a python expert here

```
Statistic,Layer 1,Layer 2,value
"['Niche overlap (D)', 'cl_prediction_maxent', 'vl_prediction_maxent', 0.651129981460896]"
"['Niche overlap (I)', 'cl_prediction_maxent', 'vl_prediction_maxent', 0.8902500062842025]"
"['Correlation coeff', 'cl_prediction_maxent', 'vl_prediction_maxent', 0.494894]"
```

### CODE REVIEW

File: src/raster/r.niche.similarity/r.niche.similarity.py [hunk 1]
```diff
@@ -88,8 +87,8 @@ def tmpname(prefix):
     """Generate a tmp name which contains prefix, store the name in the
     global list.
     """
-    tmpf = prefix + str(uuid.uuid4())
-    tmpf = string.replace(tmpf, "-", "_")
+    tmpf = 'prefix{}'.format(uuid.uuid4())
+    tmpf = tmpf.replace("-", "_")
```
Comment (petrasovaa) [inline src/raster/r.niche.similarity/r.niche.similarity.py]: Perhaps this function could be used instead:

https://grass.osgeo.org/grass82/manuals/libpython/script.html#script.utils.append_uuid

File: src/raster/r.niche.similarity/r.niche.similarity.py [hunk 2]
```diff
@@ -219,7 +218,7 @@ def main(options, flags):
 
     # Write results to csv file
     if OPF:
-        IND = [["Statistic", "Layer 1", "Layer 2", "value"]] + Dind + Iind + Cind
+        IND = [["Statistic", "Layer 1", "Layer 2", "value"], Dind, Iind, Cind]
         import csv
 
         with open(OPF, "wb") as f:
```
Comment (petrasovaa) [inline src/raster/r.niche.similarity/r.niche.similarity.py]: This should fix the other error:
```suggestion
        with open(OPF, "w") as f:
```

Comment (veroandreo) [reply src/raster/r.niche.similarity/r.niche.similarity.py]: yes, I had it locally and forgot to change it in the actual PR that I did online, thanks!

**Comment by ecodiv** [comment]
@veroandreo I edited your pull request, fixing the errors reported in https://github.com/OSGeo/grass-addons/issues/897, using the same code I have used in other addons to create temporary file names. It also fixes the printing to csv issue (by reverting to old code). For me this works, can you please check?

**Review by ecodiv — COMMENTED** [review]
See proposed changes (pushed to this branch)

**Comment by veroandreo** [comment]
works as expected, thanks for checking and fixing; merging

**Pull request merged**